### ISSUE TITLE
a few common special cases

### ISSUE THREAD

**Issue by daryanypl** [derek73/python-nameparser#57]
Sister Souljah -- "Sister" is more of a title than last name.
His Holiness the Dalai Lama -- "His Holiness" -- the two words together is the title.
Bob Jones, composer    
Bob Jones, author
Bob Jones, compositeur -- (French for composer)

Here's the code I used:
```
        name = HumanName('Sister Souljah')
        library_sort_name = u' '.join([name.first, name.middle, name.suffix, name.nickname, name.title])
        if name.last:
            library_sort_name = u''.join([name.last, ", ", library_sort_name])
        print "library_sort_name=%s" % library_sort_name
        
        name = HumanName('His Holiness the Dalai Lama')
        library_sort_name = u' '.join([name.first, name.middle, name.suffix, name.nickname, name.title])
        if name.last:
            library_sort_name = u''.join([name.last, ", ", library_sort_name])
        print "library_sort_name=%s" % library_sort_name
        
        name = HumanName('Bob Jones, author')
        library_sort_name = u' '.join([name.first, name.middle, name.suffix, name.nickname, name.title])
        if name.last:
            library_sort_name = u''.join([name.last, ", ", library_sort_name])
        print "library_sort_name=%s" % library_sort_name
        
        name = HumanName('Bob Jones, compositeur')
        library_sort_name = u' '.join([name.first, name.middle, name.suffix, name.nickname, name.title])
        if name.last:
            library_sort_name = u''.join([name.last, ", ", library_sort_name])
        print "library_sort_name=%s" % library_sort_name

```

**Comment by derek73** [comment]
I added some things to the titles constant. 

It's not clear what you would want to do with the Dalai Lama because I think technically the entire thing is more of a title, depending if you believe in reincarnation I guess. 

"Sister Souljah" is unchanged below, is it not correct?

Here's what it does now: 

```
$ python tests.py "Sister Souljah"
<HumanName : [
	title: 'Sister' 
	first: 'Souljah' 
	middle: '' 
	last: '' 
	suffix: ''
	nickname: ''
]>
$ python tests.py "His Holiness the Dalai Lama"
<HumanName : [
	title: 'His Holiness the Dalai' 
	first: '' 
	middle: '' 
	last: 'Lama' 
	suffix: ''
	nickname: ''
]>
$ python tests.py "Bob Jones, composer"
<HumanName : [
	title: 'composer' 
	first: 'Bob Jones' 
	middle: '' 
	last: '' 
	suffix: ''
	nickname: ''
]>
```

**Comment by derek73** [comment]
btw, what are you doing with `library_sort_name`? I wonder if you could do something overriding `__eq__()`?

**Comment by daryanypl** [comment]
Good question, and thanks for working on it.  I started writing the answer,
and realized I need to consult with a coworker or two who are experts on
the library science name parsing topic.  I'll email around and get back to
you.

I'm a software engineer for the New York Public Library.  I use HumanName
to convert author names from the form they're displayed in ("Bob Jones") to
the form they get sorted in ("Jones, Bob").  I also use it in a bit of code
that normalizes names before I try to determine the likelihood that two
authors from two different books are actually the same person (what's the
chance that Bob Jones and Robert L. Jones are the same person?).  The code
is in its early stages, and the data is complex.  For an example, check
http://viaf.org/viaf/search?query=local.personalNames+all+%22mark%20twain%22&sortKeys=holdingscount&recordSchema=BriefVIAF
.  VIAF gets author names from library consortia all over the world.  So
Bob Jones writes "My Memoirs", that book is owned by some libraries in the
US and let's say Netherlands.  The US consortium tells VIAF they have a
book "My Memoirs" written by Robert Jones, the Netherlands consortium tells
VIAF they have a book with the translated title "Happy Life", published
originally as "My Memoirs" by Bob Jones, Jr, with Alice Kale translator,
and then VIAF creates records for two Bobs and one Alice.  So when a
library gets books, their authors are coming in in metadata in various
formats from the reseller/open sourcer/etc., and we try our best to match
those names against each other and VIAF and let them be sortable for
library patrons.

Best,
Darya

**Comment by daryanypl** [comment]
I wrote a little script that went through the MARC21 records in VIAF, looking at section 100, field "c", which is the field where librarians write down authors' personal titles.  I wrote out the more common titles into the attached file.  The good: these are awesome, from "anarchist" to "stroke survivor".  The bad: they're a bit too much -- it would be rare that such detailed professional description is used in the author's name field.  Still, it might be the best we can do for now.

**Comment by daryanypl** [comment]
[contributor_titles_summary_1.txt](https://github.com/derek73/python-nameparser/files/800675/contributor_titles_summary_1.txt)

**Comment by derek73** [comment]
I added most of these names to the titles set and published them in v0.5.2. 

Probably most nouns could be part of a title. Probably with some NLP and ML stuff you could get pretty good at figuring out the name parts, but for now those titles all seemed pretty on par with what we already have in there so may as well add them. Nice that they came from actual data someplace instead of just me putting my thinking cap on.
